cognite/client/_http_client.py

```diff
@@ -97,13 +97,16 @@ def __init__(
         config: HTTPClientConfig,
         session: requests.Session,
         retry_tracker_factory: Callable[[HTTPClientConfig], _RetryTracker] = _RetryTracker,
+        refresh_auth_header: Callable[[MutableMapping[str, Any]], None] | None = None,
```

Comment:
Can do this and drop the null check on L131.
```suggestion
        refresh_auth_header: Callable[[MutableMapping[str, Any]], None] = lambda headers: None,
```

Reply:
Having an anonymous function as the default argument would break the "pickleability" of any `CogniteClient` with using it. I'd rather change it to a positional arg and rework all the tests 😓